src/test/java/store/controller/CategoryControllerTest.java

```diff
@@ -0,0 +1,40 @@
+package store.controller;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.springframework.beans.factory.annotation.Value;
+
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class CategoryControllerTest {
+
+    @InjectMocks
+    private CategoryController categoryController;
+
+    @Mock
+    private CategoryController controller;
+
+    @Mock
+    @Value("#{'${categories.list}'.split(',')}")
+    private List<String> categories;
+
+    @Test
+    public void testGetAll(){
+        //when
+        when(controller.getAll()).thenReturn(categories);
+
+        //then
+        List<String> resultList = categoryController.getAll();
```

Comment:
it's not necessary to make "when" for one object(contoller) and then use another object(categoryController) to check method.